shipwright/build.py

```diff
@@ -50,9 +51,27 @@ def build(client, git_rev, container):
         'rev': git_rev
     }
 
-    def process_event_(evt):
-        evt_parsed = json_loads(evt)
-        return merge(merge_config)(evt_parsed)
+    # docker-py has an issue where it doesn't handle chunked responses from
+    # Docker correctly, and so the build API yields *chunks*, rather than the
+    # valid JSON objects that it is documented as yielding.
+    # We therefore maintain a buffer and read our own valid JSON out of that.
+    # This can probably be replaced this issue is fixed - ideally we'd use the
+    # 'decode' option on client.build to receive already parsed JSON objects.
+    #   https://github.com/docker/docker-py/issues/1059
+    buffer = b''
+
+    def process_event_(buffer, data):
+        buffer += data
+        for line in buffer.split(b'\r\n'):
```

Comment:
Ok, so a `split('\n')` and a `strip`? I think the CRLF is from Docker regardless of the platform, but I don't know for sure - on my Mac it shouldn't be using CRs but it is.